internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,400 @@
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not create a new reader for blob %q: %v", blobName, err)
+	}
+	defer reader.Close()
+	return reader, nil
+}
+
+// getRevisionFromFile extracts and returns the revision of the source code from an OSS-Fuzz coverage
+// report, given the content of the source-map file (a file in the OSS-Fuzz coverage bucket that
+// links coverage build dates to the revisions of the source code used for the builds) and the project name.
+func getRevisionFromFile(fileBytes []byte, projectName string) (string, error) {
+	var payload map[string](map[string]string)
+	err := json.Unmarshal(fileBytes, &payload)
+	if err != nil {
+		return "", fmt.Errorf("could not unmarshal srcmap fileBytes into a map[string](map[string]string): %v", err)
+	}
+	// Get the revisionHash using the source-map file structure defined by OSS-Fuzz.
+	revisionHash := payload[fmt.Sprintf("/src/%s", projectName)]["rev"]
+	return revisionHash, nil
+}
+
+// parseCoverageSummary gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverageSummary(fileBytes []byte) (string, string, error) {
+	var summary CoverageSummary
+	err := json.Unmarshal(fileBytes, &summary)
+	if err != nil {
+		return "", "", fmt.Errorf("couldn't unmarshal fileBytes into a CoverageSummary: %v", err)
+	}
+	// Return branch coverage and line coverage using the coverage summary structure.
+	return formatCoverage(summary.Data[0].Totals["branches"]), formatCoverage(summary.Data[0].Totals["lines"]), nil
+}
+
+// getLogs gets the log-files list of a fuzz-target on a given day.
+// The expected date format is "YYYY-MM-DD".
+func getLogs(date string, projectName string, fuzzTarget string, fuzzEngine string, sanitizer string) (*storage.BucketHandle, []string, error) {
+	// logsBucket is the ClusterFuzz Google Cloud Storage bucket name
+	// containing the fuzzers logs for a given project.
+	logsBucket := fmt.Sprintf("%s-logs.clusterfuzz-external.appspot.com", projectName)
+	bucket, err := getBucket(logsBucket)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't get %s bucket: %v", logsBucket, err)
+	}
+	fuzzengine := strings.ToLower(fuzzEngine)
+	// relativePath is the relative path in the logsBucket where the logs of
+	// a given fuzz-target on a given day are saved.
+	relativePath := fmt.Sprintf("%s_%s_%s/%s_%s_%s/%s", fuzzEngine, projectName, fuzzTarget, fuzzengine, sanitizer, projectName, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't get blobs in %s in %s bucket: %v", relativePath, logsBucket, err)
+	}
+	return bucket, blobs, nil
+}
+
+// getFuzzStatsFromFile gets numFuzzTests and timeFuzzSeconds from a log file.
+func getFuzzStatsFromFile(lineScanner *bufio.Scanner) (int, float64, error) {
+	var timeFuzzSeconds float64
+	var numTests int
+	var err error
```

Comment:
Again I don't think this is needed. `err` is only used in the for loop.

Reply:
I changed it and used temporary variables instead.